pkg/scheduler/framework/types.go

```diff
@@ -937,7 +937,7 @@ func calculateResource(pod *v1.Pod) (Resource, int64, int64) {
 		ContainerFn: func(requests v1.ResourceList, containerType podutil.ContainerType) {
 			non0CPUReq, non0MemReq := schedutil.GetNonzeroRequests(&requests)
 			switch containerType {
-			case podutil.Containers:
+			case podutil.Containers, podutil.SidecarContainers:
 				non0CPU += non0CPUReq
 				non0Mem += non0MemReq
 			case podutil.InitContainers:
```

Comment:
this should also account for the sidecars in max calculation.

Reply:
Yes.
However, since sidecars are also init containers, and ContainerFn will be called also for init containers(https://github.com/kubernetes/kubernetes/pull/127878/files#diff-750acc9a82d57ae53d352363d4e75171dd1de0f68383750bd3eb7a4166bbb0b8R125).
The max calculation for sidecars will be included in L944-L945.

i.e. When we encounter a sidecar, we'll trigger ContainerFn twice now, one with type `initContainer`, and one with type `sidecarContainer`.

I did this because there might be some use cases which doesn't care if a initContainer is a sidecar, so they can only deal with `initContainer` in their ContainerFn.

However it seems that currently only scheduler uses ContainerFn. So it should be okay to only trigger ContainerFn with sidecar type for sidecars. I'm open for suggestions.

Reply:
> i.e. When we encounter a sidecar, we'll trigger ContainerFn twice now, one with type initContainer, and one with type sidecarContainer.

I feel this hidden logic may confuse future maintainers

Reply:
> i.e. When we encounter a sidecar, we'll trigger ContainerFn twice now,

Even if it didn't call the function twice, the calculation would still be correct, because `max(a, a+b) = a+b`, when resources are positive.

Reply:
@alculquicondor I realized this as well after making the comment. But the fact we have two calls for sidecars is still troubling

Reply:
@SergeyKanzhelev @alculquicondor 
I've added some comments for `ContainerFn` and changed to one call for sidecar, seems better?